src/daft-dsl/Cargo.toml

```diff
@@ -6,13 +6,21 @@ common-py-serde = {path = "../common/py-serde", default-features = false}
 common-resource-request = {path = "../common/resource-request", default-features = false}
 common-treenode = {path = "../common/treenode", default-features = false}
 daft-core = {path = "../daft-core", default-features = false}
+daft-io = {path = "../daft-io", default-features = false}
 daft-sketch = {path = "../daft-sketch", default-features = false}
 derive_more = {workspace = true}
 indexmap = {workspace = true}
 itertools = {workspace = true}
 pyo3 = {workspace = true, optional = true}
 serde = {workspace = true}
+serde_json = {workspace = true}
 typetag = {workspace = true}
+# Lance SDK dependencies
+lance = "0.34.0"
+arrow = "55.0"
+arrow-ipc = "55.0"
+arrow2 = {workspace = true}
+tokio = {version = "1.0", features = ["rt", "rt-multi-thread"]}
```

Comment:
Could you please move the daft-lance related functions to their own daft-functions-kv module please?